source/interact-data/specify-query.txt

```diff
@@ -484,3 +483,425 @@ as the shown in the following example:
+with the specified ``_id`` field value:
+
+.. code-block:: ruby
+
+   Band.find('6725342d4cb3e161059f91d7')
+
+.. note:: Type Conversion
+
+   The ``find()`` method performs the type conversion of the argument that
+   you pass to the type declared for the ``_id`` field in the model. By
+   default, the ``_id`` field is defined as a ``BSON::ObjectId`` type.
+
+   The preceding example is equivalent to the following code, which
+   passes an ``BSON::ObjectId`` instance as the argument to ``find()``:
+
+   .. code-block:: ruby
+      
+      Band.find(BSON::ObjectId.from_string('5f0e41d92c97a64a26aabd10'))
+
+   If you use the {+ruby-driver+} to query on the ``_id`` field,
+   ``find()`` does not internally perform the type conversion.
+
+The ``find()`` method accepts multiple arguments, or an array of arguments.
+{+odm+} interprets each argument or array element as an
+``_id`` value, and returns documents with all the specified ``_id``
+values in an array, as shown in the following example:
+
+.. literalinclude:: /includes/interact-data/query.rb
+   :start-after: start-id-query-multiple
+   :end-before: end-id-query-multiple
+   :language: ruby
+   :dedent:
+
+The ``find()`` method exhibits the following behavior:
+
+- If you provide the same ``_id`` value more than once, {+odm+}
+  returns only one document, if one exists.
+
+- {+odm+} does not return documents in an ordered way. Documents might
+  be returned in different order from the order of the provided ``_id``
+  values.
+
+- If any of the ``_id`` values are not found in the database, the result
+  depends on the value of the ``raise_not_found_error`` configuration
+  option.
+  
+  If you set the option to ``true``, ``find()`` raises a
+  ``Mongoid::Errors::DocumentNotFound`` error if any of the ``_id``
+  values are not found.
+  
+  If you set the option to ``false`` and query for a single ``_id``
+  value, ``find()`` returns ``nil`` if {+odm+} does not match any
+  document. If you pass multiple ``_id`` values and
+  some or all are not matched, the return value is an array of any documents
+  that match, or an empty array if no documents match.
+
+find() Variations
+-----------------
+
+This section describes methods that are similar to the ``find()`` method
+described in the preceding section.
+
+You can use the ``find_by()`` method to retrieve documents based on the
+provided criteria. If no documents are found, it raises an error or
+returns ``nil`` depending on how you set the ``raise_not_found_error``
+configuration option.
+
+The following code demonstrates how to use the ``find_by()`` method:
+
+.. literalinclude:: /includes/interact-data/query.rb
+   :start-after: start-query-findby
+   :end-before: end-query-findby
+   :language: ruby
+   :dedent:
+
+You can use the ``find_or_create_by()`` method to retrieve documents
+based on the provided criteria. If no documents are found, it creates
+and returns a newly persisted one.
```

Comment:
Q: What does this mean?

Reply:
create is different than initialize. Basically, create makes an instance and persists it to mongodb, while initialize just creates the instance but doesnt save it anywhere. I can make this more clear